pmm/summary.go

```diff
@@ -0,0 +1,270 @@
+
+	var Collectors = []Collector{{"Collect pmm-admin check-network output",
+		[]string{"pmm-admin", "check-network"},
+		strings.Join([]string{dirname, strings.Join([]string{"pmm-admin_check-network_", cmdHostname, ".txt"}, "")}, "/")},
+		{"Collect pmm-admin list output",
+			[]string{"pmm-admin", "list"},
+			strings.Join([]string{dirname, strings.Join([]string{"pmm-admin_list_", cmdHostname, ".txt"}, "")}, "/")},
+		{"Collect ps output",
+			[]string{"sh", "-c", "ps aux | grep exporter | grep -v grep"},
+			strings.Join([]string{dirname, strings.Join([]string{"ps_exporter_", cmdHostname, ".txt"}, "")}, "/")},
+		{"Collect pt-summary output",
+			[]string{"pt-summary"},
+			strings.Join([]string{dirname, strings.Join([]string{"pt-summary_", cmdHostname, ".txt"}, "")}, "/")},
+		{"Collect list of open ports",
+			[]string{"netstat", "-punta"},
+			strings.Join([]string{dirname, strings.Join([]string{"netstat_", cmdHostname, ".txt"}, "")}, "/")}}
+
+	switch service.Platform() {
+	case "linux-upstart":
+		Collectors = append(Collectors, Collector{"Collect service output",
+			[]string{"service", "--status-all"},
+			strings.Join([]string{dirname, strings.Join([]string{"sysv_service_", cmdHostname, ".txt"}, "")}, "/")})
+	case "linux-systemd":
+		Collectors = append(Collectors, Collector{"Collect systemctl output",
+			[]string{"systemctl", "-l", "status"},
+			strings.Join([]string{dirname, strings.Join([]string{"systemd_", cmdHostname, ".txt"}, "")}, "/")})
+	case "unix-systemv":
+		Collectors = append(Collectors, Collector{"Collect initctl output",
+			[]string{"initctl", "list"},
+			strings.Join([]string{dirname, strings.Join([]string{"upstart_", cmdHostname, ".txt"}, "")}, "/")})
+	case "darwin-launchd":
+		Collectors = append(Collectors, Collector{"Collect LaunchDaemons output",
+			[]string{"launchctl", "bslist"},
+			strings.Join([]string{dirname, strings.Join([]string{"launchd_", cmdHostname, ".txt"}, "")}, "/")})
+	}
+
+	for _, service := range CheckMonitoredDBServices() {
+		switch {
+		case (service == "mysql"):
+			Collectors = append(Collectors, Collector{"Collect pt-mysql-summary output",
+				[]string{"pt-mysql-summary"},
+				strings.Join([]string{dirname, strings.Join([]string{"pt-mysql-summary_", cmdHostname, ".txt"}, "")}, "/")})
+		case (service == "mongodb"):
+			Collectors = append(Collectors, Collector{"Collect pt-mongodb-summary output",
+				[]string{"pt-mongodb-summary"},
+				strings.Join([]string{dirname, strings.Join([]string{"pt-mongodb-summary_", cmdHostname, ".txt"}, "")}, "/")})
+		}
+	}
+
+	// Perform exec commands and collect results
+	for _, info := range Collectors {
+		summaryLogger.Printf("%s - %v\n", info.CollectorDescription, info.CollectData())
+	}
+
+	// Collect pmm logs from /var/log
+	logsDir := "/var/log"
+	srcf, err := os.Open(logsDir)
+	if err != nil {
+		fmt.Printf("Failed to open directory %s with %s\n", logsDir, err)
+		summaryLogger.Println(err)
```

Comment:
After that line, we crash on `defer srcf.Close()` below

Reply:
moved into the used else part